EssentialFeed/EssentialFeedTests/Feed API/LoadFeedImageCommentsFromRemoteUseCaseTests.swift

```diff
@@ -0,0 +1,243 @@
+		let (_, client) = makeSUT()
+		
+		XCTAssertTrue(client.requestedURLs.isEmpty)
+	}
+	
+	func test_loadImageCommentsFromURL_requestsCommentsFromURL() {
+		let imageID = UUID().uuidString
+		let url = URL(string: "https://a-given-url.com/image/\(imageID)/comments")!
+		let (sut, client) = makeSUT()
+		
+		_ = sut.loadImageComments(imageID: imageID) { _ in }
+		
+		XCTAssertEqual(client.requestedURLs, [url])
+	}
+	
+	func test_loadImageCommentsFromURLTwice_requestsCommentsFromURLTwice() {
+		let imageID = UUID().uuidString
+		let url = URL(string: "https://a-given-url.com/image/\(imageID)/comments")!
+		let (sut, client) = makeSUT()
+		
+		_ = sut.loadImageComments(imageID: imageID) { _ in }
+		_ = sut.loadImageComments(imageID: imageID) { _ in }
+		
+		XCTAssertEqual(client.requestedURLs, [url, url])
+	}
+	
+	func test_loadImageCommentsFromURL_deliversConnectivityErrorOnClientError() {
+		let (sut, client) = makeSUT()
+		let clientError = NSError(domain: "a client error", code: 0)
+		
+		expect(sut, toCompleteWith: failure(.connectivity), when: {
+			client.complete(with: clientError)
+		})
+	}
+	
+	func test_loadImageCommentsFromURL_deliversInvalidDataErrorOnNon2xxHTTPResponse() {
+		let (sut, client) = makeSUT()
+		
+		let samples = [199, 150, 300, 400, 500]
+		
+		samples.enumerated().forEach { index, code in
+			expect(sut, toCompleteWith: failure(.invalidData), when: {
+				client.complete(withStatusCode: code, data: anyData(), at: index)
+			})
+		}
+	}
+	
+	func test_loadImageCommentsFromURL_deliversInvalidDataErrorOn2xxHTTPResponseWithEmptyData() {
+		let (sut, client) = makeSUT()
+		
+		let samples = [200, 201, 299]
+		
+		samples.enumerated().forEach { index, code in
+			expect(sut, toCompleteWith: failure(.invalidData), when: {
+				let emptyData = Data()
+				client.complete(withStatusCode: code, data: emptyData, at: index)
+			})
+		}
+	}
+	
+	func test_load_deliversErrorOn2xxHTTPResponseWithInvalidJSON() {
+		let (sut, client) = makeSUT()
+		
+		let samples = [200, 201, 299]
+		
+		samples.enumerated().forEach { index, code in
+			expect(sut, toCompleteWith: failure(.invalidData), when: {
+				let invalidJSON = Data("invalid json".utf8)
+				client.complete(withStatusCode: code, data: invalidJSON, at: index)
+			})
+		}
+	}
+	
+	func test_load_deliversNoItemsOn2xxHTTPResponseWithEmptyJSONList() {
+		let (sut, client) = makeSUT()
+		
+		let samples = [200, 201, 299]
+		
+		samples.enumerated().forEach { index, code in
+			expect(sut, toCompleteWith: .success([]), when: {
+				let emptyListJSON = makeItemsJSON([])
+				client.complete(withStatusCode: code, data: emptyListJSON, at: index)
+			})
+		}
+	}
+	
+	func test_load_deliversItemsOn2xxHTTPResponseWithJSONItems() {
+		let (sut, client) = makeSUT()
+		
+		let item1 = makeItem(id: UUID(),
+							 message: "a message",
+							 createdAt: "2020-05-20T11:24:59+0000".ISO8601Date,
+							 username: "a username")
+		
+		let item2 = makeItem(id: UUID(),
+							 message: "a message",
```

Comment:
The `message` should be different than the first one to increase coverage 

```suggestion
		let item2 = makeItem(id: UUID(),
							 message: "another message",
```